vllm/config.py

```diff
@@ -2078,6 +2079,12 @@ def _get_quantization_config(
             return quant_config
         return None
 
+    def with_hf_config(self, hf_config: PretrainedConfig) -> "VllmConfig":
+        model_config = copy.deepcopy(self.model_config)
+        model_config.hf_config = hf_config
+
+        return replace(self, model_config=model_config)
```

Comment:
we should flatten `VllmConfig` in the future, `hf_config` can be the same level as `model_config` . and we can have a top-level function `vllm_config.replace_with(hf_config=hf_config)`

Reply:
Some values in `model_config` depend on the `hf_config`, so it might be better to just have a convenience property to access `hf_config` directly from `VllmConfig`, instead of changing the nested structure.